app.rb

```diff
@@ -0,0 +1,131 @@
+require_relative 'rental'
+require_relative 'classroom'
+require_relative 'teacher'
+require_relative 'person'
+require_relative 'book'
+require_relative 'student'
+
+class App
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  def start_console
+    puts 'welcome to our school Library App!'
+    until list_of_options
+      input = gets.chomp
+      if input == '7'
+        puts 'Thank You for using our school Library!'
+        break
+      end
+
+      option input
+    end
+  end
+
+  def list_all_books
+    puts 'Database is empty! Add a book.' if @books.empty?
+    @books.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_all_persons
+    puts 'Database is empty! Add a person.' if @persons.empty?
+    @persons.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, Age: #{person.age}, id: #{person.id}"
+    end
+  end
+
+  def create_person
+    print 'press 1 to create a student, press 2 to create a teacher : '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Try again'
+    end
+  end
+
+  def create_student
+    puts 'create a new student'
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      student = Student.new(age: age, name: name, parent_permission: parent_permission, classroom: @classroom)
+      @persons << student
+      puts 'Student doesn\'t have parent permission, can\'t rent books'
+    when 'y'
+      student = Student.new(age: age, name: name, parent_permission: parent_permission, classroom: @classroom)
+      @persons << student
+      puts 'Student created successfully'
+    end
+  end
+
+  def create_teacher
```

Comment:
- [ ] As suggested by the previous reviewer,  Good job so far, when entering details for a Teacher, we're expected to enter age, name, and specialization. Currently, your implementation only captures age and specialization and you have left out the name of a Teacher. Kindly add the name of a Teacher. 👍🏽

![image](https://github.com/Roland-Ntwali/School-library-app/assets/63473584/fde0fe4c-b8d5-47f8-b8c1-b493525b6c7e)